src/events/crawler/lib/fetch.js

```diff
@@ -210,3 +210,49 @@ export const getArcGISCSVURL = async function(serverNumber, dashboardId, layerNa
   const { serviceItemId } = layerMetadata;
   return `https://opendata.arcgis.com/datasets/${serviceItemId}_0.csv`;
 };
+
+/**
+ * Get the URL for the CSV data from an ArcGIS dashboard
+ * @param {*} dashboardId the ID of the dashboard
+ * @param {*} widgetName the name of the widget in the manifest
+ *
+ * The dashboardId is passed to the iframe that renders URL.  For example, 
+ *   If you have https://maps.arcgis.com/apps/opsdashboard/index.html#/ec4bffd48f7e495182226eee7962b422 is 
+ *   dashboardId = ec4bffd48f7e495182226eee7962b422
+ *
+ * To get the widgetName, load the manifest, reformat it, and search for a widget that has a caption
+ * that matches the title of table shown in the UI.  The caption is HTML but some places put images 
+ * instead of text so we should the name when scraping.  For example,
+ *    https://maps.arcgis.com/sharing/rest/content/items/ec4bffd48f7e495182226eee7962b422/data?f=json
+ * has a widget named 'COVID-19 Cases by County' which has a datasourceId = 24f4fcf164ad4b4280f08c8939dd5dc7
+ *
+ *
+ */
+export const getArcGISCSVURL_NEW = async function(dashboardId, widgetName) {
```

Comment:
Let's call it `getArcGISCSVURLByWidgetName`
```suggestion
export const getArcGISCSVURLByWidgetName = async function(dashboardId, widgetName) {
```

Reply:
Lost the function when I forked the repo.  I was originally working on a local branch.